### PR TITLE
Package ocrdesktop installation and configuration into RPM

### PR DESCRIPTION
**Pull request by kk-at-redhat**

### CODE REVIEW

File: .gitignore
Comment (vojtapolasek) [inline .gitignore:1]: Please make this as a separate PR if possible.

File: ks/vojtux_common.ks [hunk 1]
```diff
@@ -179,10 +179,6 @@ rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-rpmfusion-free-fedora-*-primary
 echo "Importing RPM Fusion keys"
 rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-rpmfusion-nonfree-fedora-*-primary
 
-#installing ocrdesktop
```
Comment (vojtapolasek) [inline ks/vojtux_common.ks:182]: Thanks for removing those lines, but please also add the new package into the %packages section of kickstart.

File: rpm/ocrdesktop/ocrdesktop
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop:1]: I think this is not the way to go. I think we should source the file from the upstream repo, not copy it here.

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Requires: python3 tesseract tesseract-langpack-eng python3-pillow python3-pyatspi python-pytesseract libwnck3 gtk3
+
+SOURCE0:  ocrdesktop
+SOURCE1: 09-keybindings-ocrdesktop
+
+%description
+RPM package for ocrdesktop
+
+%install
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:15]: Please see this file:
https://github.com/vojtapolasek/vojtux/blob/master/specs/vojtux-docs-en.spec
I believe the %install section can be omitted and the automatic macro can be used in %prep section.

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+install -d %{buildroot}/opt/ocrdesktop
+install -m 755 %{SOURCE0} %{buildroot}/opt/ocrdesktop/ocrdesktop
+install -d %{buildroot}/etc/dconf/db/local.d
+install %{SOURCE1} %{buildroot}/etc/dconf/db/local.d/09-keybindings-ocrdesktop
+
+%files
+%dir /opt/ocrdesktop
+/opt/ocrdesktop/ocrdesktop
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:23]: As we are installing it as a package, we want the file in /usr/bin.

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+/etc/dconf/db/local.d/09-keybindings-ocrdesktop
+
+%post
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:26]: If we put the file directly into /usr/bin, we can remove this section.

File: rpm/ocrdesktop/ocrdesktop
Comment (vojtapolasek) [reply rpm/ocrdesktop/ocrdesktop:1]: Also note that there is a man page in the upstream repo.

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Thank you very much for this contribution.
Please see comments.

File: .gitignore
Comment (kk-at-redhat) [reply .gitignore:1]: Moved to a separate PR as requested.

File: ks/vojtux_common.ks (hunk 1 above)
Comment (kk-at-redhat) [reply ks/vojtux_common.ks:182]: Package built in COPR and added to KS.

File: rpm/ocrdesktop/ocrdesktop
Comment (kk-at-redhat) [reply rpm/ocrdesktop/ocrdesktop:1]: It's using latest release now.

File: rpm/ocrdesktop/ocrdesktop.spec (hunk 3 above)
Comment (kk-at-redhat) [reply rpm/ocrdesktop/ocrdesktop.spec]: Changed the path to /usr/bin

File: rpm/ocrdesktop/ocrdesktop.spec (hunk 4 above)
Comment (kk-at-redhat) [reply rpm/ocrdesktop/ocrdesktop.spec]: Fixed.

File: rpm/ocrdesktop/ocrdesktop.spec (hunk 2 above)
Comment (kk-at-redhat) [reply rpm/ocrdesktop/ocrdesktop.spec:19]: Added some changes, please, let me know if that's what you had in mind.

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+%install
+install -d %{buildroot}/usr/bin
+cp ocrdesktop %{buildroot}/usr/bin/ocrdesktop
+chmod 755 %{buildroot}/usr/bin/ocrdesktop
+install -d %{buildroot}/etc/dconf/db/local.d
+install %{SOURCE1} %{buildroot}/etc/dconf/db/local.d/09-keybindings-ocrdesktop
+
+%files
+/usr/bin/ocrdesktop
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:27]:
```suggestion
%{_bindir}/ocrdesktop
```

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+/etc/dconf/db/local.d/09-keybindings-ocrdesktop
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:28]:
```suggestion
%{_sysconfdir}/dconf/db/local.d/09-keybindings-ocrdesktop
```

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+URL:      https://github.com/chrys87/ocrdesktop/
+SOURCE0:  https://github.com/chrys87/ocrdesktop/archive/refs/tags/4.0.tar.gz
+SOURCE1:  09-keybindings-ocrdesktop
+
+%description
+RPM package for ocrdesktop
+
+%prep
+%setup -q
+
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:18]: There should be a %build section even if it is empty.

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello @kk-at-redhat  and thank you for your updates.
I put there some comments especially around the %install section.
I got inspired by spec files here:
https://rpm-packaging-guide.github.io/
Also if you run the rpmlint program on the spec file, it can help you to find some warnings.
Sorry if I might seem a bit picky, but as this is one of first custom spec files, I would like to get it right so that later packages are easier to create.
Let me know what you think about my suggestions.

**Comment by kk-at-redhat** [comment]
Made changes as requested. Please, let me know if this looks better.

File: rpm/ocrdesktop/09-keybindings-ocrdesktop [hunk 8]
```diff
@@ -0,0 +1,5 @@
+[org/mate/desktop/keybindings/custom9]
+action='/opt/ocrdesktop/ocrdesktop -l eng'
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/09-keybindings-ocrdesktop:2]:
```suggestion
action='/usr/bin/ocrdesktop -l eng'
```
Hello @kk-at-redhat  and thank you for the update. You have just left this old path here, other than that it looks great! Thank you.

Comment (kk-at-redhat) [reply rpm/ocrdesktop/09-keybindings-ocrdesktop]: Thank you, fixed.

**Comment by vojtapolasek** [comment]
Hello @kk-at-redhat  and sorry for the delay. I have one remaining thing left. I think it would actually make sense to drop the dconf configuration which configures the keybinding and make it as a separate package or something like that... That means that the ocrdesktop package will install only the ocrdesktop software, nothing else.
We can then create extra packages (in spearate PRs), e.g. ocrdesktop-eng, which would depend on Tesseract English language data and it would add keyboard shortcut specific for English. We can do something similar for Czech or other language. What do you think about this suggestion?

**Review by vojtapolasek — APPROVED** [review]

**Comment by vojtapolasek** [comment]
Hello @kk-at-redhat  and thank you, I think all feedback has been addressed. Please just rebase so that I can merge it.

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+%install
+install -d %{buildroot}%{_bindir}
+install -m 0755 ocrdesktop %{buildroot}%{_bindir}/ocrdesktop
+install -d %{buildroot}%{_sysconfdir}/dconf/db/local.d
+install %{SOURCE1} %{buildroot}%{_sysconfdir}/dconf/db/local.d/09-keybindings-ocrdesktop
+
+%files
+%{_bindir}/ocrdesktop
+%{_sysconfdir}/dconf/db/local.d/09-keybindings-ocrdesktop
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:31]: I suggest we change this because we no longer perform local configuration, we ship a package.
```suggestion
%{_sysconfdir}/dconf/db/distro.d/09-keybindings-ocrdesktop
```

File: rpm/ocrdesktop/ocrdesktop.spec [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+%prep
+%setup -q
+
+%build
+
+%install
+install -d %{buildroot}%{_bindir}
+install -m 0755 ocrdesktop %{buildroot}%{_bindir}/ocrdesktop
+install -d %{buildroot}%{_sysconfdir}/dconf/db/local.d
```
Comment (vojtapolasek) [inline rpm/ocrdesktop/ocrdesktop.spec:26]:
```suggestion
install -d %{buildroot}%{_sysconfdir}/dconf/db/distro.d
```

File: rpm/ocrdesktop/ocrdesktop.spec (hunk 9 above)
Comment (kk-at-redhat) [reply rpm/ocrdesktop/ocrdesktop.spec]: Changed.

File: ks/vojtux_common.ks [hunk 11]
```diff
@@ -182,14 +184,6 @@ rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-rpmfusion-free-fedora-*-primary
 echo "Importing RPM Fusion keys"
 rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-rpmfusion-nonfree-fedora-*-primary
 
-# import Vojtux-apps key
```
Comment (vojtapolasek) [inline ks/vojtux_common.ks:185]: Hello, something probably went wrong during the rebase... we don't want this part to be removed, please add it back or remove the commit which removes it.

Comment (kk-at-redhat) [reply ks/vojtux_common.ks:185]: Sorry about that. Added it back.

**Review by vojtapolasek — APPROVED** [review]
Thank you very much, looks good now.
